subprojects/plugins/src/main/java/org/gradle/api/tasks/bundling/War.java

```diff
@@ -166,4 +174,17 @@ public void setWebXml(@Nullable File webXml) {
         this.webXml = webXml;
     }
 
+    /**
+     * Returns the app directory of the task. Defaults to {@code src/main/webapp}.
+     *
+     * @return The app directory.
+     * @since 7.1
+     */
+    @Incubating
+    @Optional
+    @PathSensitive(PathSensitivity.RELATIVE)
+    @InputFiles
+    public DirectoryProperty getWebAppDir() {
```

Comment:
I know it will make things more annoying, but we may want to call this `getWebAppDirectory` to further distinguish it from the old convention property that was just a File/string.

This is just a suggestion.